server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -331,7 +332,13 @@ public void dispatchBadRequest(final RestChannel channel, final ThreadContext th
             } else {
                 e = new ElasticsearchException(cause);
             }
-            channel.sendResponse(new RestResponse(channel, BAD_REQUEST, e));
+            // unless it's a http headers validation error, we consider any exceptions encountered so far during request processing
+            // to be a problem of invalid/malformed request (hence the RestStatus#BAD_REQEST (400) HTTP response code)
+            if (e instanceof HttpHeadersValidationException) {
```

Comment:
Does the HttpHeadersValidationException wraps the ElasticsearchSecurityException for normal authc failure ? Does that change the error code or message seen on auth failure ?